### DISCUSSION TITLE
Trouble understanding the map

### DISCUSSION THREAD

**Discussion by denilsonsa** [divVerent/aaaaxy#336] [Q&A] [answered]
I understand the map is a graph of nodes (rooms) and edges.

A green checkmark shows all notes have been collected in those "rooms".

A dark gray node shows dead-ends.

But I still don't understand these:

* Why are some nodes shaking?
    * I can see "Leap of Faith", "The Hub", "The Anti-Hub".
* Why are some nodes blinking white?
    * In my case, "It Only Takes One".
* Why are some edges a moving white dashed pattern?
    * In my case, one edge from "The One Way Round" to "Get It Back!", and another edge from "Berlin Road Rage" to "The Torus".
* Why sometimes I can't move the selection towards a certain node (using the gamepad/keyboard), but I can click on it to reach it?
* Why I could use one of the L or R gamepad buttons/triggers to move the selection to a different far away node?

**Comment by divVerent** [comment, marked as answer]
- Shaking nodes are "hubs" you will probably need to return at some point.
- The blinking white node is where the end of the game lies.
- Edges with moving white pattern are edges you have not taken yet between two checkpoints you have already visited.
  - Getting this from The One Way Round to Get It Back seems highly unusual. Wonder how you managed to do that? It requires you to reach Get It Back without having touched The One Way Round right before, which should be impossible.
- If you can't reach a node with the keyboard, you are not following the path to it. You should be able to reach it by keyboard somehow.
- The escape button and some other buttons jump to "leap of faith", the very first checkpoint. This exists to ensure you can never get stuck on an "island" even if you manage to sequence break in a way the game can't find your checkpoint path.

**Comment by denilsonsa** [comment]
> Wonder how you managed to do that?

No clue. Here's the current status of my map:

https://github.com/divVerent/aaaaxy/assets/121676/15afff1b-d5a5-4f17-8265-ca6540d87186

Please keep in mind it has been months since I last played the game, and many more months since I explored the majority of it. I don't remember much anymore.

**Reply by divVerent** [reply]
Ah... I vaguely remember there once was a time where The One Way Round had two checkpoints with the same name; this could have led to the skip you see. Also, it might have been possible to touch A New Beginning and then turn back before getting to Get It Back.

Anyway - the best way to fix the keyboard-ability of the map is to go to A New Beginning and go via The One Way Round to Get It Back, which will complete the missing edge.

I will later investigate if I can find any remaining sequence break that fails to connect The One Way Round to Get It Back.

The other not yet done edges were optional and thus what you see is expected.

**Reply by divVerent** [reply]
Also, I just added code to allow following dashed edges by keyboard, as there really was no reason to not allow that.

**Reply by denilsonsa** [reply]
> I just added code to allow following dashed edges by keyboard

I hope it also works by gamepad. ;)

**Reply by divVerent** [reply]
Definitely, gamepad and keyboard invoke the exact same code there :)
